media-gfx/blender/blender-4.0.2.ebuild

```diff
@@ -0,0 +1,488 @@
+		-DWITH_CYCLES_DEVICE_ONEAPI=no
+		-DWITH_CYCLES_DEVICE_CUDA=$(usex cuda)
+		-DWITH_CYCLES_DEVICE_HIP=no
+		-DWITH_CYCLES_DEVICE_OPTIX=$(usex optix)
+		-DWITH_CYCLES_EMBREE=$(usex embree)
+		-DWITH_CYCLES_HIP_BINARIES=no
+		-DWITH_CYCLES_ONEAPI_BINARIES=no
+		-DWITH_CYCLES_OSL=$(usex osl)
+		-DWITH_CYCLES_PATH_GUIDING=$(usex openpgl)
+		-DWITH_CYCLES_STANDALONE=no
+		-DWITH_CYCLES_STANDALONE_GUI=no
+		-DWITH_DOC_MANPAGE=$(usex man)
+		-DWITH_FFTW3=$(usex fftw)
+		-DWITH_GHOST_WAYLAND=$(usex wayland)
+		-DWITH_GHOST_WAYLAND_APP_ID="blender-${BV}"
+		-DWITH_GHOST_WAYLAND_DBUS=$(usex wayland)
+		-DWITH_GHOST_WAYLAND_DYNLOAD=no
+		-DWITH_GHOST_WAYLAND_LIBDECOR=no
+		-DWITH_GHOST_X11=$(usex X)
+		-DWITH_GMP=$(usex gmp)
+		-DWITH_GTESTS=$(usex test)
+		-DWITH_HARU=$(usex pdf)
+		-DWITH_HEADLESS=$($(use X || use wayland) && echo OFF || echo ON)
+		-DWITH_INSTALL_PORTABLE=no
+		-DWITH_IMAGE_OPENEXR=$(usex openexr)
+		-DWITH_IMAGE_OPENJPEG=$(usex jpeg2k)
+		-DWITH_IMAGE_WEBP=$(usex webp)
+		-DWITH_INPUT_NDOF=$(usex ndof)
+		-DWITH_INTERNATIONAL=$(usex nls)
+		-DWITH_JACK=$(usex jack)
+		-DWITH_MATERIALX=no
+		-DWITH_MEM_JEMALLOC=$(usex jemalloc)
+		-DWITH_MEM_VALGRIND=$(usex valgrind)
+		-DWITH_MOD_FLUID=$(usex fluid)
+		-DWITH_MOD_OCEANSIM=$(usex fftw)
+		-DWITH_NANOVDB=$(usex nanovdb)
+		-DWITH_OPENAL=$(usex openal)
+		-DWITH_OPENCOLLADA=$(usex collada)
+		-DWITH_OPENCOLORIO=$(usex color-management)
+		-DWITH_OPENIMAGEDENOISE=$(usex oidn)
+		-DWITH_OPENMP=$(usex openmp)
+		-DWITH_OPENSUBDIV=$(usex opensubdiv)
+		-DWITH_OPENVDB=$(usex openvdb)
+		-DWITH_OPENVDB_BLOSC=$(usex openvdb)
+		-DWITH_POTRACE=$(usex potrace)
+		-DWITH_PUGIXML=$(usex pugixml)
+		-DWITH_PULSEAUDIO=$(usex pulseaudio)
+		-DWITH_PYTHON_INSTALL=no
+		-DWITH_DRACO=no
+		-DWITH_PYTHON_INSTALL_NUMPY=no
+		-DWITH_PYTHON_INSTALL_ZSTANDARD=no
+		-DWITH_SDL=$(usex sdl)
+		-DWITH_STATIC_LIBS=no
+		-DWITH_STRICT_BUILD_OPTIONS=yes
+		-DWITH_SYSTEM_EIGEN3=yes
+		-DWITH_SYSTEM_FREETYPE=yes
+		-DWITH_SYSTEM_LZO=yes
+		-DWITH_TBB=$(usex tbb)
+
+		-DWITH_USD=no
+		-DWITH_HYDRA=no
+
+		-DWITH_XR_OPENXR=no
+	)
+
+	if use optix; then
+		mycmakeargs+=(
+			-DCYCLES_RUNTIME_OPTIX_ROOT_DIR="${EPREFIX}"/opt/optix
+			-DOPTIX_ROOT_DIR="${EPREFIX}"/opt/optix
+		)
+	fi
+
+	# This is currently needed on arm64 to get the NEON SIMD wrapper to compile the code successfully
+	use arm64 && append-flags -flax-vector-conversions
+
+	append-cflags $(usex debug '-DDEBUG' '-DNDEBUG')
+	append-cppflags $(usex debug '-DDEBUG' '-DNDEBUG')
+
+	if tc-is-gcc ; then
+		# These options only exist when GCC is detected.
+		# We disable these to respect the user's choice of linker.
+		mycmakeargs+=(
+			-DWITH_LINKER_GOLD=no
+			-DWITH_LINKER_LLD=no
+		)
+		# Ease compiling with required gcc similar to cuda_sanitize but for cmake
+		use cuda && use cycles-bin-kernels && mycmakeargs+=( -DCUDA_HOST_COMPILER="$(cuda_gccdir)" )
+	fi
+	if tc-is-clang ; then
+		mycmakeargs+=(
+			-DWITH_CLANG=yes
```

Comment:
With `USE="-X openexr openimageio opencolorio dds tbb -oidn opengl alembic bullet collada -embree ffmpeg fftw fluid jack jemalloc jpeg2k man nanovdb ndof iodn openal -openpgl opensubdiv openvdb -optix -cuda -osl pdf potrace pulseaudio wayland"` and clang configure fails with
```
-- Found LLVM: /usr/lib/llvm/17/lib64/libLLVM-17.0.6.so  
-- Could NOT find Clang (missing: _CLANG_LIBRARIES) 
CMake Error at build_files/cmake/macros.cmake:1533 (message):
  Clang required but not found
Call Stack (most recent call first):
  build_files/cmake/platform/platform_unix.cmake:493 (set_and_warn_library_found)
  CMakeLists.txt:1304 (include)
```
I suggest instead of `if tc-is-clang` use `-DWITH_LLVM=$(usex osl) -DWITH_CLANG=$(usex osl)`, because for me it looks like they are needed directly only for OSL 
https://github.com/blender/blender/blob/v4.0.2/CMakeLists.txt#L1121-L1124